lxd/snapshot_common.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"fmt"
+	"gopkg.in/robfig/cron.v2"
+	"strings"
+	"time"
+)
+
+// SnapshotScheduleAliases contains the mapping of scheduling aliases to cron syntax
+var SnapshotScheduleAliases = map[string]string{
+	"@hourly":   "0 * * * *",
+	"@daily":    "0 0 * * *",
+	"@midnight": "0 0 * * *",
+	"@weekly":   "0 0 * * 0",
+	"@monthly":  "0 0 1 * *",
+	"@annually": "0 0 1 1 *",
+	"@yearly":   "0 0 1 1 *",
+}
```

Comment:
So the downside to this approach is that all instances using one of those aliases will get their snapshots happen at the exact same time which can create a lot of load. I was hoping that by adding those we'd get to fix this issue at the same time.

The plan I had in mind for that is to pass the instance or volume ID to the function figuring out the snapshot schedule.
This can be then be used to seed a random number generator with a fixed value and then populate the flexible fields of the cron schedule.

So for `@daily` we'd need to generate a unique hour and minute value derived from the instance or volume ID.
There are some example functions for that in `lxd/util/random.go`.

You could for example call `GetStableRandomInt64FromList` with the instance/volume ID as first argument and a list of in64 from 0-23 for hours and 0-59 for minutes which will get you a stable "random" value for that one instance/volume.
And then same thing for day, week, ... whatever is relevant for the given alias.

Does that make sense?

Reply:
@ it does :)